plugins/heroku/src/tasks/review.ts

```diff
@@ -27,13 +28,17 @@ options:
         throw error
       }
 
+      await setStageConfigVars(this.logger, 'review', 'production', this.options.pipeline)
+
       const pipeline: HerokuApiResPipeline = await herokuClient.get(`/pipelines/${this.options.pipeline}`)
 
-      const reviewAppId = await getHerokuReviewApp(this.logger, pipeline.id)
+      let reviewAppId = await getHerokuReviewApp(this.logger, pipeline.id)
 
-      writeState('review', { appId: reviewAppId })
+      if (!reviewAppId) {
+        reviewAppId = await buildHerokuReviewApp(this.logger, pipeline.id)
+      }
```

Comment:
**question**: if the review app already exists, does updating the stage config update the app's config?

Reply:
No, it doesn't 😞 . And worse, because it's the push to the branch that triggers heroku to rebuild, it's too quick to apply (either by stage or by app) via circleci. 

Trying to work out how much of a problem this is. If it's just review-apps, the following conditions are present:
-  a PR has been raised and a review app has been created
- one of more conf vars that are needed for the postbuild steps have been updated since the review app has been created. 
- there has been a push to the branch since the conf vars have been updated
 
is it enough to add some info in the error message if the build fails? I can't see how n-gage could avoid this either?

If this also applies to staging as well, then it's a bigger problem! I'll investigate.